packages/inter-protocol/src/auction/scheduler.js

```diff
@@ -116,17 +117,29 @@ export const computeRoundTiming = (params, baseTime) => {
  * @property {() => void} startRound
  */
 
+/**
+ * @typedef {object} ScheduleNotification
+ *
+ * @property {Timestamp | undefined} activeStartTime start time of current
+ *    auction if auction is active
+ * @property {Timestamp | undefined} nextStartTime start time of next auction
+ * @property {Timestamp | undefined} nextDescendingStepTime if auction is
+ *    active we're not on the last step, when the next descending step will be
```

Comment:
are there really 2^3 combinations of defined/undefined?   in particular, why can a notification be empty?

the possible values seem to depend on `auctionState` but not in a way that's clear to me. Please add some conditions or assertions in `publishSchedule` relating it to the state.